extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-sra.js

```diff
@@ -0,0 +1,438 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AmpA4A,
+  RENDERING_TYPE_HEADER,
+  XORIGIN_MODE,
+} from '../../../amp-a4a/0.1/amp-a4a';
+import {Services} from '../../../../src/services';
+import {
+  AmpAdNetworkDoubleclickImpl,
+  getNetworkId,
+  constructSRABlockParameters,
+  TFCD,
+  resetSraStateForTesting,
+} from '../amp-ad-network-doubleclick-impl';
+import {
+  MANUAL_EXPERIMENT_ID,
+} from '../../../../ads/google/a4a/traffic-experiments';
+import {
+  EXPERIMENT_ATTRIBUTE,
+} from '../../../../ads/google/a4a/utils';
+import {utf8Encode} from '../../../../src/utils/bytes';
+import {BaseElement} from '../../../../src/base-element';
+import {createElementWithAttributes} from '../../../../src/dom';
+import {layoutRectLtwh} from '../../../../src/layout-rect';
+import {Xhr, FetchResponseHeaders} from '../../../../src/service/xhr-impl';
+import {dev} from '../../../../src/log';
+import '../../../amp-ad/0.1/amp-ad';
+import * as sinon from 'sinon';
+
+const config = {amp: true, allowExternalResources: true};
+
+/**
+  Need to set allowExternalResources to true, because otherwise when the iframes
+  for testing are created, their underlying resources are not actually loaded,
+  and most importantly for iframes their srcdoc gets set, which breaks
+  calling setAttribute('src', 'foo') on the iframe, which will cause all these
+  tests to fail.
+*/
+describes.realWin('amp-ad-network-doubleclick-impl', config , env => {
+  let sandbox;
+  let doc;
+
+  beforeEach(() => {
+    doc = env.win.document;
+    sandbox = env.sandbox;
+    // ensures window location == AMP cache passes
+    env.win.AMP_MODE.test = true;
+  });
+
+  function createAndAppendAdElement(opt_attributes, opt_type, opt_domElement) {
+    const element = createElementWithAttributes(
+        doc, opt_type || 'amp-ad',
+        opt_attributes || Object.assign(
```

Comment:
Actually you don't need opt_attributes || here.  Object assign will copy whenever is in opt_attributes over your default object.  This way you could just provide {width: 51} as your opt_attributes and you'll get {type: 'doubleclick', height: 30, width: 51}